src/index.js

```diff
@@ -86,105 +86,89 @@ class WarmUP {
    * */
   afterDeployFunctions () {
     this.configPlugin()
-    if (this.warmup.prewarm) {
+
+    if (this.warmupOpts.prewarm) {
       return this.warmUpFunctions()
     }
   }
 
+  getGlobalConfig (possibleConfig, defaultOpts = {}) {
+    const folderName = (typeof possibleConfig.folderName === 'string') ? possibleConfig.folderName : '_warmup'
+    const pathFolder = path.join(this.serverless.config.servicePath, folderName)
+
+    return {
+      folderName,
+      pathFolder,
+      pathFile: `${pathFolder}/index.js`,
+      pathHandler: `${folderName}/index.warmUp`,
+      cleanFolder: (typeof possibleConfig.cleanFolder === 'boolean') ? possibleConfig.cleanFolder : defaultOpts.cleanFolder,
+      name: (typeof possibleConfig.name === 'string') ? possibleConfig.name : defaultOpts.name,
+      role: (typeof possibleConfig.role === 'string') ? possibleConfig.role : defaultOpts.role,
+      tags: (typeof possibleConfig.tags === 'object') ? possibleConfig.tags : defaultOpts.tags,
+      schedule: (typeof possibleConfig.schedule === 'string') ? [possibleConfig.schedule]
+        : (Array.isArray(possibleConfig.schedule)) ? possibleConfig.schedule : defaultOpts.schedule,
+      memorySize: (typeof possibleConfig.memorySize === 'number') ? possibleConfig.memorySize : defaultOpts.memorySize,
+      timeout: (typeof possibleConfig.timeout === 'number') ? possibleConfig.timeout : defaultOpts.timeout,
+      prewarm: (typeof possibleConfig.prewarm === 'boolean') ? possibleConfig.prewarm : defaultOpts.prewarm
+    }
+  }
+
+  getFunctionConfig (possibleConfig, defaultOpts) {
+    if (typeof possibleConfig === 'undefined') {
+      return defaultOpts
+    }
+
+    if (typeof possibleConfig !== 'object') {
+      return Object.assign({}, defaultOpts, { enabled: possibleConfig })
+    }
+
+    // Keep backwards compatibility for now
```

Comment:
This is the backward compatibility little hack 😄

Reply:
Yes it helps if we don't want to bump major.